core/domain/state_domain.py

```diff
@@ -916,8 +940,1085 @@ def get_all_outcomes(self) -> List[Outcome]:
+                            rule_value, True, True
+                        )
+                    except Exception:
+                        rule_value = rule_spec.inputs['x']
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type equals \'contains\' string '
+                            f'value - \'{rule_value}\''
+                        )
+
+                if rule_spec.rule_type == 'IsLessThan':
+                    try:
+                        rule_value = float(rule_spec.inputs['x'])
+                        self._set_lower_and_upper_bounds(
+                            range_var, lower_infinity,
+                            rule_value, False, False
+                        )
+                    except Exception:
+                        rule_value = rule_spec.inputs['x']
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsLessThan\' contains string '
+                            f'value - {rule_value}'
+                        )
+
+                if rule_spec.rule_type == 'IsWithinTolerance':
+                    rule_value_x = rule_spec.inputs['x']
+                    rule_value_tol = rule_spec.inputs['tol']
+                    try:
+                        rule_value_x = float(rule_value_x)
+                        rule_value_tol = float(rule_value_tol)
+                        # `tol` value in `IsWithinTolerance` must be positive.
+                        if rule_value_tol <= 0.0:
+                            raise utils.ValidationError(
+                                f'The rule \'{rule_spec_index}\' of answer '
+                                f'group \'{ans_group_index}\' having '
+                                f'rule type \'IsWithinTolerance\' '
+                                f'have \'tol\' value less than or equal to '
+                                f'zero in NumericInput interaction.'
+                            )
+                        self._set_lower_and_upper_bounds(
+                            range_var, rule_value_x - rule_value_tol,
+                            rule_value_x + rule_value_tol, True, True
+                        )
+                    except Exception:
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' having rule '
+                            f'type \'IsWithinTolerance\' contains string '
+                            f'value - \'{rule_value_x}\', \'{rule_value_tol}\''
+                        )
+
+                if rule_spec.rule_type == 'IsGreaterThan':
```

Comment:
put is greater than below the is less than part

Reply:
Done